### PR TITLE
Multiple UI improvements

### PR DESCRIPTION
**Pull request by asaadmahmoodspin**
Hey guys have a look.
Changes include:
- Adding box-shadow for modals.
- Fixing invite member inputs on mobile.
- Improving the alignment of the members count in the channel header.
- Fixing word wrapping in modals.
- PLT-2880 - Adding recent mentions icon to the RHS on mobile.
- PLT-2988 - Fixing the tutorials on mobile.
- Fixed some stuff with the compact view like, colon positioning after username, post spacing, images spacing for sequential messages etc.

### CODE REVIEW

File: webapp/components/sidebar.jsx [hunk 1]
```diff
@@ -370,6 +370,13 @@ export default class Sidebar extends React.Component {
         this.setState({showDirectChannelsModal: false});
     }
 
+    openLeftSidebar() {
+        if (Utils.isMobile()) {
+            $('.app__body .inner-wrap').toggleClass('move--right');
```
Comment (enahum) [inline webapp/components/sidebar.jsx]: can we try and do this without jquery?

File: webapp/components/sidebar_right_menu.jsx [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
 // See License.txt for license information.
 
+import $ from 'jquery';
```
Comment (enahum) [inline webapp/components/sidebar_right_menu.jsx]: same here

File: webapp/components/tutorial/tutorial_tip.jsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
 // See License.txt for license information.
 
+import $ from 'jquery';
```
Comment (enahum) [inline webapp/components/tutorial/tutorial_tip.jsx]: and here

File: webapp/components/sidebar.jsx (hunk 1 above)
Comment (asaadmahmoodspin) [inline webapp/components/sidebar.jsx]: Not sure, we're already doing it with jquery in other parts of the app. If someone knows how to do it just let me know and I'll change it.

**Comment by asaadmahmoodspin** [comment]
Can we get someone to review this PR? If anyone has suggestion on another way to do this then let me know or maybe we can take this and then a dev can go ahead and maybe remove all the jquery stuff we use to toggle the sidebars.

File: webapp/components/sidebar.jsx (hunk 1 above)
Comment (enahum) [inline webapp/components/sidebar.jsx]:
```javascript
document.querySelector('.app__body .inner-wrap').classList.toggle('move--right');
document.querySelector('.app__body .sidebar--left').classList.toggle*'move--right');
```

File: webapp/components/sidebar_right_menu.jsx [hunk 4]
```diff
@@ -60,13 +64,83 @@ export default class SidebarRightMenu extends React.Component {
...
+            term: terms,
+            do_search: true,
+            is_mention_search: true
+        });
+    }
+
+    closeLeftSidebar() {
+        if (Utils.isMobile()) {
+            setTimeout(() => {
+                $('.app__body .inner-wrap').removeClass('move--right');
```
Comment (enahum) [inline webapp/components/sidebar_right_menu.jsx]: to remove a class
```javascript
document.querySelectror('.app__body .inner-wrap').classList.remove('move--right');
```

File: webapp/components/sidebar_right_menu.jsx [hunk 5]
```diff
@@ -60,13 +64,83 @@ export default class SidebarRightMenu extends React.Component {
...
+                $('.app__body .sidebar--left').removeClass('move--right');
+            });
+        }
+    }
+
+    openRightSidebar() {
+        if (Utils.isMobile()) {
+            setTimeout(() => {
+                $('.app__body .inner-wrap').addClass('move--left-small');
```
Comment (enahum) [inline webapp/components/sidebar_right_menu.jsx]: To add a class

```javascript
document.querySelector('.app__body .inner-wrap').classList.add('move--left-small');
```

**Comment by enahum** [comment]
+1

**Comment by hmhealey** [comment]
+1 I think we need to spend some time thinking about how we're going to do stuff like this if we fully remove jquery. There's no point really in removing it if we just duplicate the functionality with plain JS. The point is more that we're trying to have the view state like whether or not the sidebars are open controlled more by React and less with CSS classes getting applied and removed in event handlers.

**Comment by asaadmahmoodspin** [comment]
Agreed.

**Pull request merged**